BDMS 389: well inventory csv updates

jacob-a-brown opened the pull request:
<!--
- Title the PR with the JIRA project & ticket number and short description (e.g., BDMS-123: Example display bug), or NO TICKET
- Keep sections short and scannable.
-->
🚨 I added the feature file **well-inventory-csv.feature** to this PR so that the tests can be run against it 🚨 

### **Why**
_This PR addresses the following problem / context:_
- Fields that have been added for the implementation of **well-core-information.feature**, **well-additional-information.feature**, and **well-notes.feature** need to be handled by the `/well-inventory-csv` endpoint and the feature tests in **well-inventory-csv.feature**. 

### **How**
_Implementation summary - the following was changed / added / removed:_
- New fields were mapped in both `CreateWell` and `CreateThing` schemas. They were placed in the former if water well-specific and added to the latter if they apply to all `thing_types`
- All models that relate to a `Thing`, such as `MonitoringFrequencyHistory` and `MeasuringPointHistory`, `Notes`, or `DataProvenance` are now handled in the function **services/thing_helper.py::add_thing**. All of this functionality was moved there to keep the code DRY since this is also used in POST `/thing/water-well` (and eventually other `thing_types`).
  - Models that are dependent on the creation of objects from other models, like `PermissionHistory`, are not handled by this function.
  - `add_thing` handles the creation of `LocationThingAssociation` and `GroupThingAssociation` already, so that logic was removed from **api/well_inventory.py**
- The `WellTransferer` was updated to exclude fields as necessary.
- The csv tables used for the tests were updated to use restricted vocabulary from the lexicon.

### **Notes**
_Any special considerations, workarounds, or follow-up work to note?_
- Not every field in the `WellInventoryRow` schema has a home in a model. I marked them with the inline comment `TODO: needs a home`. Should I create new fields for these? If so, some of them can probably be in the polymorphic notes table (`target_table=contact`), but others I'm not too sure. Thoughts on those fields?
  - Please make comments about them in **schemas/well_inventory.py** where I made the inline comments.
- The default port for the DB engine was changed to 54321 to correspond with the new Docker setup.
- Am I handling `LocationThingAssociation.effective_start` correctly in `add_thing`?
- @ksmuczynski, @chasetmartin and I determined that inventorying a well is a field activity, so a `FieldActivity` record is added too (and if a water level is determined/chemistry sample taken at the same time they can all be part of the same field event)

jacob-a-brown committed:
Merge branch 'staging' into bdms-389-well-inventory-csv-updates

jacob-a-brown committed:
fix: use lexicon values in well inventory CSV testing data

update the CSV files to use values restricted by the lexicon

jacob-a-brown committed:
feat: ignore .DS_Store

jacob-a-brown committed:
refactor: update origin_source to origin_type in lexicon

origin_source is freeform, whereas origin_type is a list of
pre-defined values

jacob-a-brown committed:
feat: add well inventory csv gherkin file

jacob-a-brown committed:
refactor: default engine's port to 54321 to reflect docker

The docker compose file was changed to map Postgres to host port 54321.

jacob-a-brown committed:
feat: add well inventory as possible activity_type

jacob-a-brown committed:
note: indicate which fields still need a home in the models

These fields were noted with the inline comment
"TODO: needs a home"

jacob-a-brown committed:
feat: update CreateWell and CreateThing schemas for well inventory CSV import

Both optional and required fields have been added to the CreateWell and CreateThing
schemas per the well inventory CSV import requirements. The fields added
to CreateThing are applicable to all thing types, while the fields added to
CreateWell are specific to well things.

jacob-a-brown committed:
feat/refactor: move logic for thing tables to add_thing

The function add_thing should handle all of the data in CreateWell,
so that it can be used in multiple places without duplicating code.

jacob-a-brown committed:
feat: update well transfer script to account for updated CreateWell schema

jacob-a-brown committed:
Merge branch 'well-inventory-csv' into bdms-389-well-inventory-csv-updates

jacob-a-brown committed:
feat: add field activity record for the well inventory

there can be multiple activities per field event, one of which
is the well inventory

jirhiker commented on db/engine.py:
```diff
@@ -109,7 +109,7 @@ def getconn():
     password = os.environ.get("POSTGRES_PASSWORD", "")
     host = os.environ.get("POSTGRES_HOST", "localhost")
-    port = os.environ.get("POSTGRES_PORT", "5432")
+    port = os.environ.get("POSTGRES_PORT", "54321")
```
54321 is only for testing and should not be default

jacob-a-brown replied:
Because `5432` is used within Docker, but `54321` is used for testing, I set `POSTGRES_PORT=5432` in docker-compose.yml and then updated my environment file to use the correct port for testing.

I also updated `.env.example` to have the other environment variables in use (`POSTGRES_PORT` and `REBUILD_DB`)

jirhiker reviewed (changes requested):
looks good. Only issue is the port assignment

jacob-a-brown committed:
fix: reset default POSTGRES_PORT to 5432 and update POSTGRES_PORT in docker-compose.yml

Inside Docker the app needs to use port 5432 to connect to Postgres, but
on the host machine we want to use 54321. This can be set in the
.env file, but to prevent 54321 from being used within Docker
we set POSTGRES_PORT to 5432.
